components/movementsensor/merged/merged.go

```diff
@@ -0,0 +1,376 @@
+	if err != nil {
+		return err
+	}
+
+	m.angVel, err = firstGoodSensorWithProperties(
+		deps, newConf.AngularVelocity, m.logger,
+		&movementsensor.Properties{AngularVelocitySupported: true}, "angular_velocity")
+	if err != nil {
+		return err
+	}
+
+	m.linAcc, err = firstGoodSensorWithProperties(
+		deps, newConf.LinearAcceleration, m.logger,
+		&movementsensor.Properties{LinearAccelerationSupported: true}, "linear_acceleration")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (m *merged) Position(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	if m.pos == nil {
+		return geo.NewPoint(math.NaN(), math.NaN()), math.NaN(),
+			movementsensor.ErrMethodUnimplementedPosition
+	}
+	return m.pos.Position(ctx, extra)
+}
+
+func (m *merged) Orientation(ctx context.Context, extra map[string]interface{}) (spatialmath.Orientation, error) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	if m.ori == nil {
+		nanOri := spatialmath.NewOrientationVector()
+		nanOri.OX = math.NaN()
+		nanOri.OY = math.NaN()
+		nanOri.OZ = math.NaN()
+		nanOri.Theta = math.NaN()
+		return nanOri,
+			movementsensor.ErrMethodUnimplementedOrientation
+	}
+	return m.ori.Orientation(ctx, extra)
+}
+
+func (m *merged) CompassHeading(ctx context.Context, extra map[string]interface{}) (float64, error) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	if m.compass == nil {
+		return math.NaN(),
+			movementsensor.ErrMethodUnimplementedCompassHeading
+	}
+	return m.compass.CompassHeading(ctx, extra)
+}
+
+func (m *merged) LinearVelocity(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	if m.linVel == nil {
+		return r3.Vector{X: math.NaN(), Y: math.NaN(), Z: math.NaN()},
+			movementsensor.ErrMethodUnimplementedLinearVelocity
+	}
+	return m.linVel.LinearVelocity(ctx, extra)
+}
+
+func (m *merged) AngularVelocity(ctx context.Context, extra map[string]interface{}) (spatialmath.AngularVelocity, error) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	if m.angVel == nil {
+		return spatialmath.AngularVelocity{X: math.NaN(), Y: math.NaN(), Z: math.NaN()},
+			movementsensor.ErrMethodUnimplementedAngularVelocity
+	}
+	return m.angVel.AngularVelocity(ctx, extra)
+}
+
+func (m *merged) LinearAcceleration(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	if m.linAcc == nil {
+		return r3.Vector{X: math.NaN(), Y: math.NaN(), Z: math.NaN()},
+			movementsensor.ErrMethodUnimplementedLinearAcceleration
+	}
+	return m.linAcc.LinearAcceleration(ctx, extra)
+}
+
+func mapWithSensorName(name string, accMap map[string]float32) map[string]float32 {
+	for k, v := range accMap {
+		// if map key hasn't been appended, delete
+		delete(accMap, k)
+		// if map key has already been appended, delete
+		delete(accMap, name+"_"+k)
+		// append map with the senosr name and accuracy field
+		accMap[name+"_"+k] = v
```

Comment:
Mutating the map you're currently iterating over can have very weird behavior. In many languages this invalidates the iterator in the for loop, but [in Go](https://go.dev/ref/spec#For_statements), 
> if a map entry is created during iteration, that entry may be produced during the iteration or may be skipped. The choice may vary for each entry created and from one iteration to the next.

Instead of trying to wade through this thicket, just create a new map and return that instead:
```go
result := map[string]float32{}
for k, v := range accMap {
    result[name + "_" + k] = v
}
return result
```

Reply:
That's why I'm checking the keys and deleting them first actually. It's not a simple append operation.

Reply:
Here's the situation I want to avoid:
1. You start with the name "foo" and the map {"length": 1.0}.
2. The first iteration through the loop, `k` is "length" and `v` is 1.0.
3. We delete `k` from the map, which is now {}.
4. We delete "foo_length" from the map, which is still {}.
5. We add "foo_length" to the map, which is now {"foo_length": 1.0}
6. We go to the second iteration of the loop. `k` is now "foo_length", and `v` is 1.0 again.
7. We delete `k` from the map, which is now {}.
8. We delete "foo_foo_length" from the map, which is still {}.
9. We add "foo_foo_length" to the map, which becomes {"foo_foo_length": 1.0}
10. We go to the third iteration, where `k` is now "foo_foo_length". We never exit the for loop because we keep adding the same prefix to the key over and over.

Reply:
what's adding foo_foo_length? 

I'll chaneg it I just don't see how that happens.

Reply:
The for loop iterates over all keys in the map. If you add new keys during the for loop, it might iterate over those, too.